src/FillAnySureVisitor.cpp

```diff
@@ -20,38 +20,42 @@ bool FillAnySureVisitor::Visit(Grid &ioGrid) const{
         TripleHolder fillableTriplet;
 
         for(int main_col=0; main_col<3;++main_col) {
-                // vertical
+                // go through columns
                 
-                for(int i=0; i<3; ++i) {
-                        // work with the ith region in this row
+                for(int main_row=0; main_row<3; ++main_row) {
+                        // go through rows
+
+                        // set the working region in regions[0]
                         regions = std::vector<RegionHolder>();
-                        regions.push_back(ioGrid.getRegion(3*((i+0)%3) + main_col));
-                        regions.push_back(ioGrid.getRegion(3*((i+1)%3) + main_col));
-                        regions.push_back(ioGrid.getRegion(3*((i+2)%3) + main_col));
+                        regions.push_back(ioGrid.getRegion(3*((main_row+0)%3) + main_col));
+                        regions.push_back(ioGrid.getRegion(3*((main_row+1)%3) + main_col));
+                        regions.push_back(ioGrid.getRegion(3*((main_row+2)%3) + main_col));
                         
                         for(int value = 1; value<=9; ++value) {
                                 // test for all different values
-                                //
+                                
+                                // if value is in working region skip it
                                 if(regions[0].isValuePresent(value)){continue;}
 
-                                // value is not present in the ith region
-                                // find the col where it should be
+                                // find the col where it can be
                                 
                                 possibleColumns = vector<bool>(3, false);
-                                position = regions[1].valuePosition(value);
```

Comment:
Still have the same semantic problem.
the index where we are are going to have true are the one where it is not possible to put the value.

What it reads here is : 
At the beginning, we can't put the numbers in any columns (possible is false) , but if the number is present in one column, mark it as possible.

Either rename your vector impossibleColumns, or make it store True by default, and set to false when it becomes impossible